skt/__init__.py

```diff
@@ -506,7 +512,24 @@ def mktgz(self, clean=True):
         mk = subprocess.Popen(args,
                               stdout=subprocess.PIPE,
                               stderr=subprocess.STDOUT)
-        (stdout, stderr) = mk.communicate()
+        stdout_fd = mk.stdout.fileno()
+        stdout = ""
+        while stdout_fd > -1:
+            ret = select.select([stdout_fd], [], [], 10)
+            for fd in ret[0]:
+                if fd == stdout_fd:
+                    data = os.read(fd, 1024)
+                    if not data:
+                        stdout_fd = -1
+                        mk.stdout.close()
+                    sys.stdout.write(data)
+                    stdout += data.decode('utf-8', 'replace')
```

Comment:
Hmm, why do we need to use `select` for reading from a single file descriptor and not doing anything else? I would have thought you can just read it in a loop, can't you?